src/app/shared/components/filters-list/user-radius-set/user-radius-set.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, OnInit } from '@angular/core';
+import { ChangeContext } from '@angular-slider/ngx-slider';
+import { FormControl } from '@angular/forms';
+import { Store } from '@ngxs/store';
+import {
+  GetFilteredWorkshops,
+  SetRadiusSize
+} from '../../../store/filter.actions';
+import { DestroyableDirective } from '../../../directives/destroyable.directive';
+import { debounceTime, takeUntil } from 'rxjs/operators';
+import { MatSnackBar } from '@angular/material/snack-bar';
+
+@Component({
+  selector: 'app-user-radius-set',
+  templateUrl: './user-radius-set.component.html',
+  styleUrls: ['./user-radius-set.component.scss']
+})
+export class UserRadiusSetComponent
+  extends DestroyableDirective
```

Comment:
don't that it is so relevant to create a separate directive that has only one default onDestroy method, it can be handled by the component, why did you decide to do it this way? 
cause it is a class that extends, so all other components can extend only one another class, but some components extend more important functionality, also some other components have another functionality on destroy, so people (students) can forget to call super.
let's keep the approach that already is applied on our project

Reply:
Shure, some classes can't extend it, but the most of components haven't any class for extend, and this directive will do the code smaller.
I thought that it a good point for students a bit train with use super and overrride (with simple examples in onDestroy), so i decide to do it this way, just for some new components (not for refactoring old)
But yes, there is too many old components, so it wasn't a really good idea, i was tired, sorry)